### PR TITLE
feat(styles): add delta horizon theming support for Button

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue

Part of #3213 

## Description

Button horizon support & minor component refactoring.

_Added TODO will be resolved in the next pr._

### CODE REVIEW

File: src/styles/segmented-button.scss [hunk 1]
```diff
@@ -1,75 +1,52 @@
...
-.fd-segmented-button+()
+.fd-segmented-button
     .fd-button
 */
 $block: #{$fd-namespace}-segmented-button;
-
-@mixin styleButtonContent() {
-  @content;
-}
+$block-button: #{$fd-namespace}-button;
```
Comment (InnaAtanasova) [inline src/styles/segmented-button.scss:10]: Same as the comment above about violating the self-contained rules

File: src/styles/tool-header.scss [hunk 2]
```diff
@@ -133,7 +133,8 @@ $fd-tool-header-object-status-semantic-values: (
     color: var(--sapShell_TextColor);
   }
 
-  .#{$block}__button {
+  /** TODO: Remove after button styles refactoring */
+  .#{$fd-namespace}-button.#{$block}__button {
```
Comment (InnaAtanasova) [inline src/styles/tool-header.scss:137]: You can't do this

File: src/styles/tool-header.scss [hunk 3]
```diff
@@ -160,7 +161,7 @@ $fd-tool-header-object-status-semantic-values: (
       }
     }
 
-    &-text {
+    .#{$block}__button-text {
```
Comment (InnaAtanasova) [inline src/styles/tool-header.scss:164]: is the specificity needed here?

File: src/styles/button-split.scss [hunk 4]
```diff
@@ -1,155 +1,164 @@
...
+    > .#{$block-button}--negative {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Reject_TextColor));
+    }
+  }
+
+  &--attention {
+    @include fd-button-pad(var(--sapButton_Attention_Background));
+
+    > .#{$block-button}--attention {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Attention_TextColor));
```
Comment (InnaAtanasova) [inline src/styles/button-split.scss:148]: Lots of repetitive code that can be avoided. 
For example, you can move all the specific values like this(for all types of buttons with modifier classes:
```
$button-states: (
  "ghost": (
    "btnPad": var(--fdButton_Ghost_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_TextColor)
  ),
  "positive": (
    "color": var(--sapButton_Accept_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_Accept_TextColor)
  ),
  "negative": (
    "color": var(--sapButton_Reject_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_Reject_TextColor)
  ),
  ..... other types
);

```

and then loop through them:
```
@each $set-name, $props-set in $button-states {
  &--#{$set-name} {
    color: map_get($props-set, "color");

    @include fd-button-pad(map_get($props-set, "color"));

    > .#{$block-button}--negative {
      @include fd-split-button-separator(map_get($props-set, "btnSeparatorDisplay"), map_get($props-set, "btnSeparatorBackground"));
    }
  }
}
```

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/styles/tool-header.scss (hunk 3 above)
Comment (platon-rov) [reply src/styles/tool-header.scss:164]: Yea, at least till we refactor button component, `.fd-button.fd-button--emphasized` => `.fd-button--emphasized`

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
Code looks good!
Some visual problems:
1. The offset of the focus is not correct (in my opinion). The outer "border" that is former from the offset(the outer blue border) should be the same width as the focus outline (white border). 
Fund-styles:
Screen Shot 2022-04-04 at 4 26 19 PM
Screen Shot 2022-04-04 at 4 27 58 PM

Specs:
Screen Shot 2022-04-04 at 4 26 31 PM

2. Semantic buttons have the wrong "down" state:
fund-styles:
Screen Shot 2022-04-04 at 4 29 32 PM

specs:
Screen Shot 2022-04-04 at 4 29 46 PM

Same is observed in all semantic types.

3. Toggle Hover & Down state is not observed in Fund-styles implementation
Screen Shot 2022-04-04 at 4 35 38 PM

4. Toggled disabled state is wrong. 
Fund-styles:
Screen Shot 2022-04-04 at 4 36 21 PM
Specs:
Screen Shot 2022-04-04 at 4 38 51 PM

5. HCB & HCW for Emphasized button: the text should be bold and the borders are thicker than the rest of the buttons:
Fund-styles:
Screen Shot 2022-04-04 at 4 47 47 PM

Specs:
Screen Shot 2022-04-04 at 4 48 00 PM

**Review by g-cheishvili — CHANGES_REQUESTED** [review]
As usual - nice code
I noticed segmented button shadows are bit cut off
cut off

**Comment by platon-rov** [comment]
> 1. The offset of the focus is not correct (in my opinion). The outer "border" that is former from the offset(the outer blue border) should be the same width as the focus outline (white border).

No, this is appropriate, focus-offset is 3px. Sent a link to the wiki in the Slack.

The rest, points 2-6 are fixed.

**Review by aberikashvili — APPROVED** [review]
LGTM

**Comment by platon-rov** [comment]
@InnaAtanasova okay, i changed outline offset from 3px to 4px for horizon morning & evening, it looks more accordingly to the images in the specs.

**Review by InnaAtanasova — APPROVED** [review]

**Review by g-cheishvili — APPROVED** [review]
Looks good to me

**Pull request merged**